express-notes-api/server.js

```diff
@@ -0,0 +1,155 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+const app = express();
+app.use(express.json());
+
+// errors array
+const errorArr = [
+  { error: 'Id must be a positive integer.' },
+  { error: 'Cannot find note with id.' },
+  { error: 'Content cannot be empty.' },
+  { error: 'An unexpected error occurred.' },
+];
+
+async function parse() {
+  const dataJson = await readFile('data.json', 'utf8');
+  const data = JSON.parse(dataJson);
+  return data;
+}
+
+// read all entries function and route
+async function readAll() {
```

Comment:
I see. You are pulling out the logic so that you don't have the "read all" logic and the express req/res handling all in one function. Nice.

I really like the attempt at making this clean. However, it's hard to argue that the net result is more readable. Now, if `readAll`, `readOne`, etc., were in a different file then it would make a lot of sense and would be a great separation.

So, great attempt. It's awesome seeing you thinking this way.